server_extensions_api/src/main/java/com/generalbytes/batm/server/coinutil/LnurlUtil.java

```diff
@@ -0,0 +1,65 @@
+package com.generalbytes.batm.server.coinutil;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public class LnurlUtil {
+    public static final String WITHDRAW_PATH = "withdraw";
+    public static final String WITHDRAW_CONFIRM_PATH = "withdraw-confirm";
+
+    public final String baseUrl = getBaseUrlConfiguration();
+    // some wallets will display the hostname, e.g. <hostname> is taking too long to pay [...] please contact <hostname>.
+
+    public String getLnurlQrcode(String rid, String uuid, BigDecimal cryptoAmount) {
```

Comment:
Parameters here and in getLnurlQrcode don't look like optional ones since you set them to `lnurl`. It would be better to check them too

Reply:
fixed